plugin/advisor/catalog/state.go

```diff
@@ -128,6 +128,149 @@ type databaseState struct {
 
 	context *FinderContext
 }
+
+// HasNoTable returns true if the current database has no table.
+func (d *databaseState) HasNoTable() bool {
+	for _, schema := range d.schemaSet {
+		if len(schema.tableSet) != 0 {
+			return false
+		}
+	}
+
+	return true
+}
+
+// DatabaseName returns the database name.
+func (d *databaseState) DatabaseName() string {
+	return d.name
+}
+
+// IndexFind is for find index.
+type IndexFind struct {
+	SchemaName string
+	TableName  string
+	IndexName  string
+}
+
+func (d *databaseState) FindIndex(find *IndexFind) (string, *Index) {
+	// needMatchTable is used for PostgreSQL. In PostgreSQL, the index name is unique in a schema, not a table.
```

Comment:
This comment is a bit confusing.

d.dbType != db.Postgres suggests needMatchTable is NOT used for PostgreSQL